pkg/codegen/nodejs/gen.go

```diff
@@ -1102,27 +1157,55 @@ func (mod *modContext) genFunction(w io.Writer, fun *schema.Function) (functionF
 		}
 		argsig = fmt.Sprintf("args%s: %sArgs, ", optFlag, title(name))
 	}
-	fmt.Fprintf(w, "export function %s(%sopts?: pulumi.InvokeOptions): Promise<%s> {\n",
-		name, argsig, functionReturnType(fun))
+
+	funReturnType := mod.functionReturnType(fun)
+
+	fmt.Fprintf(w, "export function %s(", name)
+	if fun.MultiArgumentInputs {
+		for _, prop := range fun.Inputs.Properties {
+			if prop.IsRequired() {
+				fmt.Fprintf(w, "%s: ", prop.Name)
+				fmt.Fprintf(w, "%s, ", mod.typeString(prop.Type, false, nil))
+			} else {
+				fmt.Fprintf(w, "%s?: ", prop.Name)
+				// since we already applied the ? to the type, we can simplify
+				// the optional-ness of the type
+				switch propType := prop.Type.(type) {
+				case *schema.OptionalType:
+					fmt.Fprintf(w, "%s, ", mod.typeString(propType.ElementType, false, nil))
+				default:
+					fmt.Fprintf(w, "%s, ", mod.typeString(propType, false, nil))
+				}
+			}
+		}
+	} else {
+		fmt.Fprintf(w, "%s", argsig)
+	}
+
+	fmt.Fprintf(w, "opts?: pulumi.InvokeOptions): Promise<%s> {\n", funReturnType)
 	if fun.DeprecationMessage != "" && mod.compatibility != kubernetes20 {
 		fmt.Fprintf(w, "    pulumi.log.warn(\"%s is deprecated: %s\")\n", name, escape(fun.DeprecationMessage))
 	}
 
 	// Zero initialize the args if empty and necessary.
-	if fun.Inputs != nil && argsOptional {
+	if fun.Inputs != nil && argsOptional && !fun.MultiArgumentInputs {
 		fmt.Fprintf(w, "    args = args || {};\n")
 	}
 
+	fmt.Fprint(w, "\n")
 	// If the caller didn't request a specific version, supply one using the version of this library.
-	fmt.Fprintf(w, "\n")
 	fmt.Fprintf(w, "    opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts || {});\n")
-
+	invokeCall := runtimeInvokeFunction(fun)
 	// Now simply invoke the runtime function with the arguments, returning the results.
-	fmt.Fprintf(w, "    return pulumi.runtime.invoke(\"%s\", {\n", fun.Token)
+	fmt.Fprintf(w, "    return pulumi.runtime.%s(\"%s\", {\n", invokeCall, fun.Token)
 	if fun.Inputs != nil {
 		for _, p := range fun.Inputs.Properties {
 			// Pass the argument to the invocation.
 			body := fmt.Sprintf("args.%s", p.Name)
+			if fun.MultiArgumentInputs {
+				body = fmt.Sprintf("%s", p.Name)
```

Comment:
Why are we doing this instead of just assigning?
```suggestion
				body = p.Name
```